### PR TITLE
[example] Yolov5 on Face Mask Detection

### PR DESCRIPTION
**Pull request by KexinFeng**
Relevant to PR https://github.com/awslabs/atlearn/pull/9 in ATLearn.

### CODE REVIEW

File: examples/docs/object_detection.md [hunk 1]
```diff
@@ -43,3 +43,23 @@ Your output should look like the following:
 An output image with bounding box will be saved as build/output/detected-dog_bike_car.png:
 
 ![detected-dogs](img/detected-dog_bike_car.png)
+
+## Run object detection example with other engines
+For objection detection application, other than the default model zoo with the default engine, we can also run it with other engines and model zoo. Here, we demonstrate with a pre-trained *YOLOV5s ONNX* model.
+
+First, the following line needs to be added into the dependencies in `build.gradle`. Otherwise, the ONNX model zoo won't be found.
```
Comment (frankfliu) [inline examples/docs/object_detection.md:50]: Since we added to `build.gradle` we don't need mention this is readme

File: examples/build.gradle [hunk 2]
```diff
@@ -32,6 +32,7 @@ dependencies {
     runtimeOnly project(":engines:pytorch:pytorch-model-zoo")
     runtimeOnly project(":engines:tensorflow:tensorflow-model-zoo")
     runtimeOnly project(":engines:mxnet:mxnet-model-zoo")
+    runtimeOnly project(":engines:onnxruntime:onnxruntime-engine")
```
Comment (frankfliu) [inline examples/build.gradle:35]: Add to `pom.xml` as well

**Review by frankfliu — CHANGES_REQUESTED** [review]

File: api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java [hunk 3]
```diff
@@ -302,23 +304,31 @@ public void drawBoundingBoxes(DetectedObjects detections) {
             g.setStroke(new BasicStroke(stroke));
             g.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
 
-            int imageWidth = image.getWidth();
-            int imageHeight = image.getHeight();
+            float widthScale = (float) (image.getWidth() / 640.0);
```
[the remark this answers is not in the document]
Comment (KexinFeng) [reply api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java:307]: Solved

File: examples/docs/mask_detection.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+In this example, the trained model is exported to ONNX file and is then also used for inference in python, which will serve as a benchmark.
+
+## Setup guide
+
+To configure your development environment, follow [setup](../../docs/development/setup.md).
+
+## Run mask detection example
+
+### Input image file
+You can find the image used in this example in the project test resource folder: `src/test/resources/maksssksksss627.png`.
```
[the remark this answers is not in the document]
Comment (KexinFeng) [reply examples/docs/mask_detection.md]: Solved.

File: examples/docs/mask_detection.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+![dogs](../src/test/resources/face_mask.png)
+
+### Build the project and run
+Use the following command to run the project:
+
+```
+cd examples
+./gradlew run -Dai.djl.default_engine=PyTorch -Dmain=ai.djl.examples.inference.MaskDetectionOnnx
```
[the remark this answers is not in the document]
Comment (KexinFeng) [reply examples/docs/mask_detection.md]: Solved. No need to specify the engine

File: examples/src/main/java/ai/djl/examples/inference/MaskDetection.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                        .optModelUrls(modelPath)
+                        .optEngine("OnnxRuntime")
+                        .optArguments(arguments)
+                        .optProgress(new ProgressBar())
+                        .build();
+
+        try (ZooModel<Image, DetectedObjects> model = criteria.loadModel()) {
+            try (Predictor<Image, DetectedObjects> predictor = model.newPredictor()) {
+                DetectedObjects detection = predictor.predict(img);
+                String outputDir = "/Users/fenkexin/Desktop/djl/examples/build/output";
```
[the remark this answers is not in the document]
Comment (KexinFeng) [reply examples/src/main/java/ai/djl/examples/inference/MaskDetection.java]: Solved

File: examples/docs/object_detection.md (hunk 1 above)
Comment (KexinFeng) [reply examples/docs/object_detection.md]: Solved

File: examples/build.gradle (hunk 2 above)
Comment (KexinFeng) [reply examples/build.gradle:35]: Solved

File: examples/src/main/java/ai/djl/examples/inference/MaskDetection.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    public static void main(String[] args) throws IOException, ModelException, TranslateException {
+        DetectedObjects detection = MaskDetection.predict();
+        logger.info("{}", detection);
+    }
+
+    public static DetectedObjects predict() throws IOException, ModelException, TranslateException {
+        Path imageFile = Paths.get("src/test/resources/maksssksksss627.png");
+        Image img = ImageFactory.getInstance().fromFile(imageFile);
+
+        String modelPath = "src/test/resources/mask.onnx";
```
[the remark this answers is not in the document]
Comment (KexinFeng) [reply examples/src/main/java/ai/djl/examples/inference/MaskDetection.java]: Solved

**Review by frankfliu — APPROVED** [review]

**Pull request merged**